src/printer.js

```diff
@@ -758,10 +762,10 @@ function genericPrintNoParens(path, options, print) {
 
       var isParentForLoop = namedTypes.ForStatement.check(parentNode) ||
         namedTypes.ForInStatement.check(parentNode) |
```

Comment:
yeah, I just pushed a fix that makes it ||. Both | and || happen to be doing roughly the same thing on boolean inputs so it still passed the tests and did what was expected.